src/include/common/utils.h

```diff
@@ -54,5 +54,24 @@ constexpr To safeIntegerConversion(From val) {
     return val;
 }
 
+template<typename T, size_t N1, size_t N2>
+constexpr std::array<T, N1 + N2> arrayConcat(const std::array<T, N1>& arr1,
+    const std::array<T, N2>& arr2) {
+    std::array<T, N1 + N2> ret;
+    std::copy_n(arr1.cbegin(), arr1.size(), ret.begin());
+    std::copy_n(arr2.cbegin(), arr2.size(), ret.begin() + arr1.size());
+    return ret;
+}
+
+template<typename Func, typename... Types, size_t... indices>
+void paramPackFuncHelper(const Func& func, std::index_sequence<indices...>, Types&&... values) {
+    ((func(indices, values)), ...);
+}
+
+template<typename Func, typename... Types>
+void paramPackFunc(const Func& func, Types&&... values) {
+    paramPackFuncHelper(func, std::index_sequence_for<Types...>(), std::forward<Types>(values)...);
+}
```

Comment:
Move these to a new/different header so that we're not pulling `utils.h` and `numeric_utils.h` into almost every file

Reply:
updated